src/Compilers/CSharp/Portable/Symbols/Synthesized/InjectedEmbeddedAttributeSymbol.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// Makes the Microsoft.CodeAnalysis.EmbeddedAttribute available in every compilation.
+    /// </summary>
+    internal sealed class InjectedEmbeddedAttributeSymbol : InjectedAttributeSymbol
+    {
+        private InjectedEmbeddedAttributeSymbol(
+            AttributeDescription description,
+            NamespaceSymbol containingNamespace,
+            CSharpCompilation compilation,
+            Func<CSharpCompilation, NamedTypeSymbol, DiagnosticBag, ImmutableArray<MethodSymbol>> getConstructors)
+            : base(description, containingNamespace, compilation, getConstructors)
+        {
+        }
+
+        public static InjectedEmbeddedAttributeSymbol Create(NamespaceSymbol containingNamespace)
+        {
+            return new InjectedEmbeddedAttributeSymbol(AttributeDescription.CodeAnalysisEmbeddedAttribute, containingNamespace, containingNamespace.DeclaringCompilation, makeConstructor);
+
+            ImmutableArray<MethodSymbol> makeConstructor(CSharpCompilation compilation, NamedTypeSymbol containingType, DiagnosticBag diagnostics)
+            {
+                return ImmutableArray.Create<MethodSymbol>(new EmbeddedAttributeConstructorSymbol(containingType));
+            }
+        }
+
+        internal override AttributeUsageInfo GetAttributeUsageInfo()
+            => new AttributeUsageInfo(validTargets: AttributeTargets.All, allowMultiple: false, inherited: false);
+
+        internal override DiagnosticBag GetDiagnostics()
+            => _diagnostics;
```

Comment:
>=> _diagnostics [](start = 12, length = 15)

Consider avoiding exposing "private" DiagnosticBag to external consumers. That can be achieved by add a DiagnosticBag parameter where the diagnostics should be added.